src/lib/tour-box/TourBox.component.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { StyleProp, Text, TextStyle, ViewStyle } from "react-native";
+
+import { RenderProps } from "../SpotlightTour.context";
+
+import { FooterContainer, MainContainer, NavButton, TitleText } from "./TourBox.styles";
+
+interface TourBoxProps extends RenderProps {
+  backText?: string;
+  nextText?: string;
+  title?: string;
+  hideNext?: boolean;
+  hideBack?: boolean;
+  onBack?: () => void;
+  onNext?: () => void;
+  backStyle?: StyleProp<ViewStyle>;
+  nextStyle?: StyleProp<ViewStyle>;
+  titleStyle?: StyleProp<TextStyle>;
+  descriptionStyle?: StyleProp<TextStyle>;
+  style?: StyleProp<ViewStyle>;
+}
+
+export const TourBox: React.FC<TourBoxProps> = props => {
+  const {
+    backText,
+    nextText,
+    title,
+    hideNext,
+    hideBack,
+    onBack,
+    onNext,
+    backStyle,
+    nextStyle,
+    titleStyle,
+    descriptionStyle,
+    style,
+    children,
+    isLast,
+    isFirst,
+    previous,
+    stop,
+    next
+  } = props;
+
+  const handleBack = () => {
+    isFirst ? stop() : previous();
+    onBack?.();
+  };
+
+  const handleNext = () => {
+    isLast ? stop() : next();
+    onNext?.();
+  };
+
+  return (
+    <MainContainer style={style}>
+      {title && (
+        <TitleText style={titleStyle}>
+          {title}
+        </TitleText>
+      )}
+
+      {typeof children === "string"
+        ? <Text style={descriptionStyle}>{children}</Text>
+        : children
+      }
```

Comment:
I think it's great to use the `children` for the body of the TourBox. But since we don't have a `description` prop, the `descriptionStyle` prop can be a little confusing. Maybe we can leave the styling to the end-user since they could pass a component or a styled component to it. What do you think? 🙂

Reply:
That is true, it can be confusing. I think only allowing the addition of children would be good here. It is more likely to pass a component with the styling.